### PR TITLE
Python 3.10 Fix jump offsets opcodes

### PR DESCRIPTION
**Pull request by yeandy**
Fixes https://github.com/apache/beam/issues/21671

The issue was caused by a deliberate [change in behavior](https://github.com/python/cpython/issues/93060#issuecomment-1133791913) in Python 3.10 due to [bpo-27129](https://bugs.python.org/issue?@action=redirect&bpo=27129). The offsets were originally byte (8 bits) offsets in Python 3.9, but instruction (16 bits) offsets in Python 3.10. The opcode numbers were halved, so to compensate, I had to double them again when reading it.

A few other tests were failing too. I had to add an opcode due to [bpo-43683](https://bugs.python.org/issue?@action=redirect&bpo=43683) in which a [new bytecode GEN_START](https://github.com/python/cpython/issues/87849) was added. 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by yeandy** [comment]
R: @robertwb

**Comment by yeandy** [comment]
retest this please

**Comment by yeandy** [comment]
R: @AnandInguva

File: sdks/python/apache_beam/typehints/trivial_inference.py [hunk 1]
```diff
@@ -376,6 +376,14 @@ def infer_return_type_func(f, input_types, debug=False, depth=0):
   inst_size = 2
   opt_arg_size = 0
 
+  # Python 3.10: bpo-27129 changes jump offsets to use instruction offsets,
+  # not byte offsets. The offsets were halved (16 bits fro instructions vs 8
+  # bits for bytes), so we have to double the value of arg.
+  if (sys.version_info.major, sys.version_info.minor) == (3, 10):
```
Comment (tvalentyn) [inline sdks/python/apache_beam/typehints/trivial_inference.py:382]: should we use the 2 multiplier for all minor versions >= 10 ?

Comment (yeandy) [reply sdks/python/apache_beam/typehints/trivial_inference.py:382]: Yes, I'll change it. When I wrote ` == (3, 10)`, I was thinking of `GEN_START` which was added in 3.10, but then removed in 3.11.

Comment (yeandy) [reply sdks/python/apache_beam/typehints/trivial_inference.py]: Updated.

**Comment by yeandy** [comment]
Merged into the 3.10 feature branch https://github.com/apache/beam/pull/22995

**Pull request closed without merging**